### PR TITLE
Deterministic output slots for smartfridges

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
Simply put, when you take items out of a smartfridge, they are output in one of the slots in a 3x3 grid inside the fridge tile in a deterministic fashion, so that a 9u pill of cyanide you might use to escape an assignment in security will always pop out at the same slot. This hopefully reduces the right-click menu usage a fair bit.

Looks kinda like this:

image

[Changelogs]: 

:cl: Naksu
tweak: Smartfridges and chem/condimasters now output items in a deterministic 3x3 grid rather than in a huge pile in the middle of the tile.
/:cl:

### CODE REVIEW

**Comment by MoreRobustThanYou** [comment]
YOU ARE A GOD, THANK YOU

**Comment by psykzz** [comment]
Can we make this a proc on all vending type machines?

**Comment by vuonojenmustaturska** [comment]
Yeah, I'm already working on an improved commit

**Comment by vuonojenmustaturska** [comment]
Red for beaker, green for pill bottles, condensed 3x3 for others. Now the beaker won't be buried under a pile of crap.

image

**Comment by vuonojenmustaturska** [comment]
image

made pill bottles with contents pop out to the left and empty ones to the right, also made the bottles have a bit more space between them

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -506,3 +506,11 @@ Class Procs:
 	. = ..()
 	if (AM == occupant)
 		occupant = null
+
+/obj/machinery/proc/adjust_item_drop_location(atom/movable/AM)	// Adjust item drop location to a 3x3 grid inside the tile, returns slot id from 0 to 8
+	var/md5 = md5(AM.name)										// Oh, and it's deterministic too. A specific item will always drop from the same slot.
+	for (var/i in 1 to 32)
+		. += hex2num(copytext(md5,i,i+1))
```
Comment (Cyberboss) [inline code/game/machinery/_machinery.dm]: Add a compile error if DM_VERSION >= 512 so we can use string iteration eventually

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Comment by 81Naoki** [comment]
That's beautiful! I have ideas for three more machines:
- Smart slime fridge and slime processor (/obj/machinery/smartfridge/extract/preloaded // machinery/processor): These output one homogenous pile of slime extracts that are all exactly the same size.
- Plant DNA manip (/obj/machinery/plantgenes): This machine outputs seed packages and plant disks on top of each other.

**Comment by Supermichael777** [comment]
Now nerf chemists

**Comment by 81Denton** [comment]
Forgot to mention the med vending machines you find in medbay. They have 10+ different items that get output as a pile.

**Pull request merged**